AK/Statistics.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021, Tobias Christiansen <tobyase@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Math.h>
+#include <AK/QuickSort.h>
+#include <AK/StdLibExtraDetails.h>
+#include <AK/Vector.h>
+
+namespace AK {
+
+template<typename T = float>
+requires(IsArithmetic<T>) class Statistics {
+public:
+    Statistics() = default;
+    ~Statistics() = default;
+
+    void add(T const& value)
+    {
+        // FIXME: Check for an overflow
+        m_sum += value;
+        m_values.append(value);
+    }
+
+    T const sum() const { return m_sum; }
+    long double average() const { return (long double)sum() / size(); }
```

Comment:
long double seems like a bit much,
double or float should suffice
(Just a nit)

Reply:
I agree, it's a bit excessive, but `long double` is the most precise type we have, so using it seems appropriate.

Reply:
but its the most expensive on x86_64 or sse systems
maybe specialize that function that it can return T if T is a floating point value otherwise return float?
we dont need 63 bits of precission

Reply:
I think I got yakbaited by thinking about optimizing for precision too early. Ill change it to use regular `float` for now.